utils/ipfs.py

```diff
@@ -0,0 +1,48 @@
+import requests
+import io
+import re
+
+from utils.config import (
+    get_web3_storage_token,
+    get_infura_io_keys,
+)
+
+
+def upload_str_to_infura_io(text) -> str:
```

Comment:
Is explicitly infura required? Can another service be used there, or own node?

Reply:
Such a hash can be generated on different services and own node. I checked it.

I use web3.storage service in project for now. It is store file in servise node and Filecoin node. I left the infra function as an alternative option. It is currently not used, but is configured to generate the same hash as web3.storage. I can remove it from the code if need.